driver-scala/src/main/scala/org/mongodb/scala/ObservableImplicits.scala

```diff
@@ -116,17 +118,41 @@ trait ObservableImplicits {
     override def subscribe(observer: Observer[_ >: GridFSFile]): Unit = Mono.from(publisher).subscribe(observer)
   }
 
-  implicit class ToSingleObservableVoid(pub: => Publisher[Void]) extends SingleObservable[Void] {
+  /**
+   * A `GridFSUploadPublisher`` that emits
+   *
+   *   - exactly one item, if the wrapped `Publisher` does not signal an error, even if the represented stream is empty;
+   *   - no items if the wrapped `Publisher` signals an error.
+   *
+   * @param pub A `Publisher` representing a finite stream.
+   */
+  implicit class ToGridFSUploadPublisherUnit(pub: => GridFSUploadPublisher[Void]) extends GridFSUploadPublisher[Unit] {
```

Comment:
This class is why we are able to simply replace `GridFSUploadPublisher[Void]` with `GridFSUploadPublisher[Unit]` in the codebase.

Reply:
This doesn't need to be a global implicit so can be added here instead: `gridfs/package.scala`

Reply:
Done.